canon/src/store/host.rs

```diff
@@ -39,16 +35,30 @@ impl HostStore {
     pub(crate) fn put(bytes: &[u8]) -> IdHash {
         debug_assert!(bytes.len() > core::mem::size_of::<IdHash>());
         let hash = Self::hash(bytes);
-        STATIC_MAP.write().insert(hash, Vec::from(bytes));
+        STATIC_MAP.with(|m| m.borrow_mut().insert(hash, Vec::from(bytes)));
         hash
     }
 
-    pub fn hash(bytes: &[u8]) -> IdHash {
+    pub(crate) fn hash(bytes: &[u8]) -> IdHash {
         let mut state = Params::new().hash_length(32).to_state();
         state.update(bytes);
 
         let mut buf = [0u8; 32];
         buf.copy_from_slice(state.finalize().as_ref());
         buf
     }
+
+    pub(crate) fn promote_bytes(id: &Id) -> Result<Vec<u8>, CanonError> {
+        STATIC_MAP.with(|m| {
+            if let Some(vec) = m.borrow_mut().remove(&id.hash()) {
+                if id.size() == vec.len() {
+                    Ok(vec)
+                } else {
+                    Err(CanonError::InvalidEncoding)
+                }
+            } else {
+                Err(CanonError::NotFound)
+            }
+        })
```

Comment:
This might be more readable with match pattern and guards, instead of nested ifs:
```rust
        match m.borrow_mut().remove(&id.hash()) {
            Some(vec) if id.size() == vec.len() => Ok(vec),
            Some(_) => Err(CanonError::InvalidEncoding),
            None => Err(CanonError::NotFound),
        }
```